Test modal

chanha-park opened the pull request:
일단 오른쪽위의 채팅모양 버튼을 누르면 소셜 창이 뜨긴 합니다. react-dom의 [createPortal](https://react.dev/reference/react-dom/createPortal) 을 이용했습니다.
하지만 이게 flex 때문인지 height 100% 때문인지 무엇인지 오버레이 - 그러니까 기존 화면 위에 겹쳐 뜨는 것 - 이 아니라 밑에 flex box의 일원으로 뜨는데요, 음... 어찌할지 고민하다 그냥 이렇게 풀리퀘 올립니다.

chanha-park committed:
[feat] modal practice. 어떻게 겹치게..?

chanha-park committed:
[feat] Nav bar 왼쪽위 로고에/main 가는 링크 추가

nimo-my reviewed (commented):
우와 찬파님 짱이에요.... createPortal 이라는게 있는지도 처음 알았어요 ㅋㅋㅋㅋ 정말 항상 역시나 최고십니당,.., LGTM!! +) 겹치게 하는 부분은 저도 좀 더 찾아볼게요!!!! 고생하셨습니다!

nimo-my reviewed (approved):
아니 Approve를 안누르고 Comment를 눌러버렸네요.. ;ㅂ; 다시 누릅니다..!

chanha-park commented:
근데 사실 요구사항 맞추려면 생각보다 많이 고쳐야해서.... 많이 손보긴 해야할거에요
1. 메인 화면과 겹쳐서 위에 그려져야함
2. 하지만 네비게이션바랑은 겹치면 안됨
3. 메인화면 완전 꽉 채울지 vs 살짝 작게해서 바깥에 어두워지는 효과 + 바깥 클릭하면 꺼지게 할지 outside click...구현은 찾으면 많긴 하네요[1](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/hooks/use-outside-click.ts) [2](https://www.robinwieruch.de/react-hook-detect-click-outside-component/) [3](https://velog.io/@tonyk0901/TIL30-React-outside-click-%EA%B5%AC%ED%98%84%ED%95%98%EA%B8%B0)
4. 네비게이션바는 어두워져야하는가?
5. ESC 버튼 잘되는지
6. 모바일 화면
7. Tab, 스크롤 등 모달 바깥 영향주지 않게 disable
8. (최적화이슈 - 위 기능들 잘 되면 고려..) db 요청 등 정보 필요한거 있을때, 모달 켰/껐 할때마다 새로 받아오는지, 저장해둔값으로 보여주는지. re-rendering 어떤지 확인. 뭔가뭔가 react hook 을 잘 써야할것같은데 잘 모르겠군요

chanha-park committed:
[feat] modal 창 위치조정 - 다른 element 위에 겹치게

Navigation bar을 제외한 부분(main element)에 position: relative를,
모달 컴포넌트에 position:absolute를 추가했습니다. 아직 modal 창이 작게 나와 조정이 필요합니다

hdoo42 reviewed (approved):
LGTM!!
